exercises/practice/connect/.meta/src/reference/java/Connect.java

```diff
@@ -0,0 +1,270 @@
+/**
+ * To solve our exercise, we are going to model our board based on disjoint-set data structure, also called a
+ * union-find data structure or merge-find set. We are going to use two sets. One for player O and the other for
+ * player X.
+ * For additional documentation, check
+ * <a href="https://en.wikipedia.org/wiki/Disjoint-set_data_structure">Wikipedia Disjoint-set Data Structure</a>
+ * or <a href="https://algs4.cs.princeton.edu/15uf">Section 1.5</a> of
+ * <i>Algorithms, 4th Edition, A nice book on DSA</i> by Robert Sedgewick and Kevin Wayne.
+ */
+class Connect {
+
+    //The disjoint-set of player O. Each element of this array represents a field and each field is connected to itself
+    // at initialization.
+    private int[] parentO;
```

Comment:
nit: This solution is fundamentally hard to read / verify because it is using a 1D representation of fundamentally 2D data. Is there any significant reason to not use a 2D data structure like an `int[][]`? Remember that mentors (and some students) will refer to this solution as an example of how to solve it. That being said, as long as it correctly solves the exercise we can use it. I just dislike having to keep so many arrays with the exact same size.

Reply:
Thanks for mentioning this. I also had doubts about the clarity of the solution.
As mentioned in the code's comments, I found it pertinent to use a **disjoint-set** data structure to solve this problem. The aim here was to introduce the ideas of the **Union Find** algorithms to solve *dynamic connectivity* problems. This might be beneficial to the students in solving this exercise and other similar exercises such as the *Percolation* problem. These algorithms and data structures are usually implemented and taught using a 1D array. That being said, it still doesn't solve our issue of clarity specific to this exercise. So my suggestion is this; What do you think of abstracting the disjoint-set data structure in a class called **Board**, that will model the HEX's board, then call operations on this board from the **Connect** class? I think this will show a clear implementation of the Union Find algorithm in a separate class (**Board**) and how it is used to provide a solution for **Connect**.

Reply:
Yes, that sounds like a significant improvement and a better abstraction.

Reply:
Understood. Please one more question; taking into consideration the comment about the enum that should be in a file of its own, should I also add the **Board** class in a separate file? Also when adding new solution files, do we need to include their path in the **config.json**?

Reply:
Yes config.json needs to include all of the files. There's a different section for editable ones vs non-editable ones. Check the spec on the website documentation on exercises.